app/Http/Controllers/ShareTokensController.php

```diff
@@ -34,7 +34,7 @@ public function share($id, Request $request)
         $share_token = $this->createShareToken($user, $observation, $request);
 
         return $this->success(
-            "https://treesnap.org/observation/$observation?token=$share_token->value"
+            "https://treesnap.org/observation/$observation->id?token=$share_token->value"
```

Comment:
```suggestion
            "https://treesnap.org/observation/{$observation->id}?token=$share_token->value"
```
This adds variable protection so the server won't mistake this for "$observation" ->id instead of instead of "$observation->id"